sympy/geometry/ellipse.py

```diff
@@ -1362,6 +1357,46 @@ def __new__(cls, *args, **kwargs):
 
         raise GeometryError("Circle.__new__ received unknown arguments")
 
+    def object_from_equation(self):
+        """Returns a Circle object from an equation of circle given as input.
+        If the equation of the circle is ax**2 + by**2 + gx + hy + c = 0, then
+        the input has to be of the following format: ax**2 + by**2 + gx + hy + c
+
+        Examples
+        ========
+
+        >>> from sympy import Circle
+        >>> from sympy.abc import x, y
+        >>> Circle.object_from_equation(x**2 + y**2 + 3*x + 4*y - 8)
+        Circle(Point2D(-3/2, -2), 3.77491721763537)
+
+        """
+
+        from sympy.abc import x, y
```

Comment:
you can get the variables from the equation (self.free_symbols) and then check to see that one or both of the names `x` and `y` appear. Note that assumptions on the user's symbols will affect this routine since ` var('x')!=var('x', real=True)`.

Reply:
@smichr 
>you can get the variables from the equation (self.free_symbols) and then check to see that one or both of the names x and y appear. 

I don't think there's a need to do that since if only one symbol either `x` or `y` come up, later in the code, in this line :
`lead_x_coeff == lead_y_coeff and (deg_x == 2 and deg_y == 2)` , the equation will make it return the `GeometryError`.

If you wanted to explain something else (which I misinterpreted) please let me know.

Reply:
```python
>>> poly(var('x',real=True),var('x'))
Poly(x, x, domain='ZZ[x]')
>>> degree(_)
0
```

The degree is 0 because there is no `var('x')` in the expression.